public/app/features/plugins/admin/state/selectors.ts

```diff
@@ -32,11 +33,15 @@ export type PluginFilters = {
 export const selectPlugins = (filters: PluginFilters) =>
   createSelector(selectAll, (plugins) => {
     const keyword = filters.keyword ? unEscapeStringFromRegex(filters.keyword.toLowerCase()) : '';
+    const pluginsForSearch: { [key: string]: string } = getPluginDetailsForFuzzySearch(plugins);
+    const pluginsAfterFuzzySearch = fuzzySearch(keyword, Object.values(pluginsForSearch));
 
     return plugins.filter((plugin) => {
-      const fieldsToSearchIn = [plugin.name, plugin.orgName].filter(Boolean).map((f) => f.toLowerCase());
+      if (keyword && pluginsAfterFuzzySearch == null) {
+        return false;
+      }
 
-      if (keyword && !fieldsToSearchIn.some((f) => f.includes(keyword))) {
+      if (keyword && pluginsAfterFuzzySearch != null && !Object.values(pluginsAfterFuzzySearch).includes(plugin.id)) {
```

Comment:
Running `Object.values(pluginsAfterFuzzySearch)` in each loop is wasteful. There are better ways to approach this problem:

* Since the object `pluginsAfterFuzzySearch` doesn't change in this loop, get the values once outside this loop and reuse them
* Better: do not use an object for the `pluginsAfterFuzzySearch` data structure. Make it an array of plugin-ids (see my previous comment about simplifying the code) and use it directly here `pluginsAfterFuzzySearch.includes(plugin.id)`